next.config.js

```diff
@@ -14,6 +14,9 @@ const nextConfig = {
   },
   reactStrictMode: false,
   async rewrites() {
+    if (!process.env.NEXT_PUBLIC_MULTIVERSX_API) {
```

Comment:
Here I would check `MULTIVERSX_CUSTOM_API` instead of `NEXT_PUBLIC_MULTIVERSX_API` because, as in the comment in .env.example, we can still use `NEXT_PUBLIC_MULTIVERSX_API` as a custom public endpoint.

Reply:
Sorry I don't get it yet.

If I change it to `MULTIVERSX_CUSTOM_API`, this method will crash when I remove/comment out `NEXT_PUBLIC_MULTIVERSX_API` in the config, because you can't rewrite undefined to something.

Reply:
Ahh, I think I get it now, if NEXT_PUBLIC_MULTIVERSX_API  is always in use we will not run into such problems 👍

Reply:
We should do the rewrites only when `MULTIVERSX_CUSTOM_API` is set. And the `NEXT_PUBLIC_MULTIVERSX_API` should still be in use. So, for example, I should still be able to set a public API endpoint different than the one from ...elrond.com. It should be clearly stated in the README that if someone provides the `MULTIVERSX_CUSTOM_API`, then the `NEXT_PUBLIC_MULTIVERSX_API` becomes the proxy like `/api/multiversx`

----

So `NEXT_PUBLIC_MULTIVERSX_API` can be:
- `https://some-custom-public-api.com`
- `/api/multiversx`

depends on whether `MULTIVERSX_CUSTOM_API` is set or not

Reply:
I can make some changes here, and you'll review it, ok? I'll show you what I meant @janniksam

Reply:
Gotcha 👍

Reply:
It can wait. Let me know when you provide changes. I won't touch it for now :) 
I agree it can be confusing for users, but changing the variable's name and docs in README about it should do the job. Maybe also some comments in .env.example

Reply:
Maybe we could even change the name for the public var to `NEXT_PUBLIC_MULTIVERSX_API_OR_PROXY` :smile: then the private one will be `MULTIVERSX_PRIVATE_API`

Reply:
> It can wait. Let me know when you provide changes. I won't touch it for now :) I agree it can be confusing for users, but changing the variable's name and docs in README about it should do the job. Maybe also some comments in .env.example

You'll have to find the right words for it, I was never a master of words ;-)

I did start to write down the three scenarios and did smoke-test them. Seems to do what it should do right now (given you do what your told to in the README.

Please double check on your end, I have a feeling I missed something. :)

Reply:
Perfect. Looks good to me. I will do a couple of manual tests, and I'll merge it.

By the way. I merged the results parser for useScQuery. There are probably minor conflicts in the .env.example. Could you resolve it on this branch? You can merge the main branch into it. You can leave your version. I only changed the comment there. Thanks.

Reply:
Rebase should be done.